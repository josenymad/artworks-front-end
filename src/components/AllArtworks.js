import React from "react";
import "../styles/all-artworks.css";
import { Link } from "react-router-dom";
import deleteArtwork from "../requests/deleteArtwork";

const AllArtworks = ({ artworks, alert, setDeleted, setSelectedArtwork }) => {
  if (artworks && artworks.length) {
    return (
      <div className="all-artworks">
        {artworks.map((artwork) => {
          return (
            <div className="all-artworks__thumbnail" key={artwork._id}>
              <div className="all-artworks__thumbnail-container">
                <object
                  data={artwork.image}
                  className="all-artworks__thumbnail-image"
                  alt="Artwork"
                  type="application/pdf"
                >
                  {artwork.product}
                </object>
                <div className="all-artworks__thumbnail-image__foreground">
                  <a
                    href={artwork.image}
                    target="_blank"
                    rel="noreferrer"
                    className="all-artworks__thumbnail-link"
                  >
                    Full View
                  </a>
                  <Link
                    to="/update-artwork"
                    className="all-artworks__thumbnail-link"
                    onClick={() => {
                      setSelectedArtwork(artwork);
                    }}
                  >
                    Update
                  </Link>
                  <button
                    className="all-artworks__thumbnail-delete"
                    type="button"
                    onClick={() => {
                      deleteArtwork(artwork._id, setDeleted);
                    }}
                  >
                    Delete
                  </button>
                </div>
              </div>
              <ul className="all-artworks__thumbnail-list">
                <li className="all-artworks__thumbnail-company">
                  {artwork.company}
                </li>
                <li className="all-artworks__thumbnail-product">
                  {artwork.product}
                </li>
                <li className="all-artworks__thumbnail-part-number">
                  {artwork.partNumber}
                </li>
                <li className="all-artworks__thumbnail-date">
                  {new Date(artwork.date).toLocaleDateString("en-gb")}
                </li>
              </ul>
            </div>
          );
        })}
      </div>
    );
  }

  return (
    <div className="all-artworks__error">
      <p>{alert}</p>
    </div>
  );
};

export default AllArtworks;
