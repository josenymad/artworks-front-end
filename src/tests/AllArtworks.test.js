import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import AllArtworks from "../components/AllArtworks";
import mockData from "./mockData.json";

describe("All Artworks", () => {
  it("renders correctly with data", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <AllArtworks artworks={mockData} />
      </BrowserRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the right amount of artwork data lists", () => {
    render(
      <BrowserRouter>
        <AllArtworks artworks={mockData} />
      </BrowserRouter>,
    );
    const renderedArtworks = screen.getAllByRole("list");

    expect(renderedArtworks).toHaveLength(3);
  });

  it("renders error without data", () => {
    render(
      <BrowserRouter>
        <AllArtworks alert="If the artworks haven't loaded after a while there may be a problem with the server" />
      </BrowserRouter>,
    );
    const error = screen.getByText(
      "If the artworks haven't loaded after a while there may be a problem with the server",
    );

    expect(error).toBeInTheDocument();
  });
});
