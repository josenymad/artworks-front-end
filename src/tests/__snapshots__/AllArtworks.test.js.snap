// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Artworks renders correctly with data 1`] = `
<DocumentFragment>
  <div
    class="all-artworks"
  >
    <div
      class="all-artworks__thumbnail"
    >
      <div
        class="all-artworks__thumbnail-container"
      >
        <object
          alt="Artwork"
          class="all-artworks__thumbnail-image"
          data="Test Image"
          type="application/pdf"
        >
          Test Product
        </object>
        <div
          class="all-artworks__thumbnail-image__foreground"
        >
          <a
            class="all-artworks__thumbnail-link"
            href="Test Image"
            rel="noreferrer"
            target="_blank"
          >
            Full View
          </a>
          <button
            class="all-artworks__thumbnail-delete"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
      <ul
        class="all-artworks__thumbnail-list"
      >
        <li
          class="all-artworks__thumbnail-company"
        >
          Test Company
        </li>
        <li
          class="all-artworks__thumbnail-product"
        >
          Test Product
        </li>
        <li
          class="all-artworks__thumbnail-part-number"
        >
          Test Part Number
        </li>
        <li
          class="all-artworks__thumbnail-date"
        >
          14/07/2021
        </li>
      </ul>
    </div>
    <div
      class="all-artworks__thumbnail"
    >
      <div
        class="all-artworks__thumbnail-container"
      >
        <object
          alt="Artwork"
          class="all-artworks__thumbnail-image"
          data="Second Test Image"
          type="application/pdf"
        >
          Second Test Product
        </object>
        <div
          class="all-artworks__thumbnail-image__foreground"
        >
          <a
            class="all-artworks__thumbnail-link"
            href="Second Test Image"
            rel="noreferrer"
            target="_blank"
          >
            Full View
          </a>
          <button
            class="all-artworks__thumbnail-delete"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
      <ul
        class="all-artworks__thumbnail-list"
      >
        <li
          class="all-artworks__thumbnail-company"
        >
          Second Test Company
        </li>
        <li
          class="all-artworks__thumbnail-product"
        >
          Second Test Product
        </li>
        <li
          class="all-artworks__thumbnail-part-number"
        >
          Second Test Part Number
        </li>
        <li
          class="all-artworks__thumbnail-date"
        >
          21/03/2023
        </li>
      </ul>
    </div>
    <div
      class="all-artworks__thumbnail"
    >
      <div
        class="all-artworks__thumbnail-container"
      >
        <object
          alt="Artwork"
          class="all-artworks__thumbnail-image"
          data="Third Test Image"
          type="application/pdf"
        >
          Third Test Product
        </object>
        <div
          class="all-artworks__thumbnail-image__foreground"
        >
          <a
            class="all-artworks__thumbnail-link"
            href="Third Test Image"
            rel="noreferrer"
            target="_blank"
          >
            Full View
          </a>
          <button
            class="all-artworks__thumbnail-delete"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
      <ul
        class="all-artworks__thumbnail-list"
      >
        <li
          class="all-artworks__thumbnail-company"
        >
          Third Test Company
        </li>
        <li
          class="all-artworks__thumbnail-product"
        >
          Third Test Product
        </li>
        <li
          class="all-artworks__thumbnail-part-number"
        >
          Third Test Part Number
        </li>
        <li
          class="all-artworks__thumbnail-date"
        >
          10/07/2022
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
